tests/test_diffraction_DiffractionPattern.py

```diff
@@ -160,11 +161,176 @@ def test_k_values_and_k_vectors(self):
+                top_center_k_vector = rowan.rotate(
+                    view_orientation, [0, first_k_value, 0]
+                )
+                bottom_center_k_vector = rowan.rotate(
+                    view_orientation, [0, last_k_value, 0]
+                )
+                left_center_k_vector = rowan.rotate(
+                    view_orientation, [first_k_value, 0, 0]
+                )
+                right_center_k_vector = rowan.rotate(
+                    view_orientation, [last_k_value, 0, 0]
+                )
+
+                npt.assert_allclose(dp.k_vectors[center, 0], top_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[center, -1], bottom_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[0, center], left_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[-1, center], right_center_k_vector)
+
+            else:
+
+                first_k_value = (-0.5 + 1 / (2 * output_size)) * scale_factor
+                last_k_value = (0.5 - 1 / (2 * output_size)) * scale_factor
+
+                npt.assert_allclose(dp.k_values[0], first_k_value)
+                npt.assert_allclose(dp.k_values[-1], last_k_value)
+
+                first_k_vector = rowan.rotate(
+                    view_orientation, [first_k_value, first_k_value, 0]
+                )
+                last_k_vector = rowan.rotate(
+                    view_orientation, [last_k_value, last_k_value, 0]
+                )
+                top_right_k_vector = rowan.rotate(
+                    view_orientation, [first_k_value, last_k_value, 0]
+                )
+                bottom_left_k_vector = rowan.rotate(
+                    view_orientation, [last_k_value, first_k_value, 0]
+                )
+
+                npt.assert_allclose(dp.k_vectors[0, 0], first_k_vector)
+                npt.assert_allclose(dp.k_vectors[-1, -1], last_k_vector)
+                npt.assert_allclose(dp.k_vectors[0, -1], top_right_k_vector)
+                npt.assert_allclose(dp.k_vectors[-1, 0], bottom_left_k_vector)
+
+                center = output_size // 2
+                top_center_k_vector = rowan.rotate(
+                    view_orientation, [0, first_k_value, 0]
+                )
+                bottom_center_k_vector = rowan.rotate(
+                    view_orientation, [0, last_k_value, 0]
+                )
+                left_center_k_vector = rowan.rotate(
+                    view_orientation, [first_k_value, 0, 0]
+                )
+                right_center_k_vector = rowan.rotate(
+                    view_orientation, [last_k_value, 0, 0]
+                )
+
+                npt.assert_allclose(dp.k_vectors[center, 0], top_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[center, -1], bottom_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[0, center], left_center_k_vector)
+                npt.assert_allclose(dp.k_vectors[-1, center], right_center_k_vector)
```

Comment:
This whole block doesn't need to be copied twice in the if/else. Try this structure:
```python
if output_size % 2 == 0:
    first_k_value = -0.5 * scale_factor
    last_k_value = (0.5 - (1 / output_size)) * scale_factor
else:
    first_k_value = (-0.5 + 1 / (2 * output_size)) * scale_factor
    last_k_value = (0.5 - 1 / (2 * output_size)) * scale_factor

npt.assert_allclose(dp.k_values[0], first_k_value)
npt.assert_allclose(dp.k_values[-1], last_k_value)

# ... continue with all the rest of the common code
```